test/common/filesystem/filesystem_impl_test.cc

```diff
@@ -224,6 +224,40 @@ TEST_F(FileSystemImplTest, IllegalPath) {
 #endif
 }
 
+TEST_F(FileSystemImplTest, ConstructedFile) {
+  const std::string new_file_path = TestEnvironment::temporaryPath("envoy_this_not_exist");
+  ::unlink(new_file_path.c_str());
+  FilePathAndType new_file_info{Filesystem::DestinationType::File, new_file_path};
+
+  FilePtr file = file_system_.createFile(new_file_info);
+  EXPECT_EQ(file->path(), new_file_path);
+  EXPECT_EQ(file->destinationType(), Filesystem::DestinationType::File);
+}
+
+TEST_F(FileSystemImplTest, ConstructedConsoleFile) {
+  FilePathAndType new_file_info{Filesystem::DestinationType::Console, ""};
+
+  FilePtr file = file_system_.createFile(new_file_info);
+  EXPECT_FALSE(file->isOpen());
+  EXPECT_EQ(file->destinationType(), Filesystem::DestinationType::Console);
```

Comment:
Verify the file->path() points to the expected path.  Same for stderr and stdout below.

Reply:
added in 4b20da95d95619ed84a0c7ee698367acf8d882a2